plugins/analogix/fu-analogix-common.c

```diff
@@ -0,0 +1,231 @@
+					bin_buf[payload_index] = tmp;
+					checksum += tmp;
+					fw_index += 2;
+					payload_index++;
+				}
+				if (start_addr == version_addr && fw_start_addr == FLASH_OCM_ADDR) {
+					/* get ocm version */
+					img_header->fw_ver = (guint8) hex_str_to_dec ((const gchar*) &hex_hdr[fw_index - 16], 2);
+					img_header->fw_ver = img_header->fw_ver << 8;
+					img_header->fw_ver |= (guint8) hex_str_to_dec ((const gchar*) &hex_hdr[fw_index - 8], 2);
+				}
+				break;
+			case 1:
+				end_of_file = TRUE;
+				break;
+			case 2:
+				addr_mode = 2;
+				most_high_addr = (guint16) hex_str_to_dec ((const gchar*) &hex_hdr[fw_index], 4);
+				checksum = line_len + rec_type +
+					(guint8) (start_addr & 0xff) +
+					(guint8) ((start_addr >> 8)&0xff) +
+					(guint8) (most_high_addr & 0xff) +
+					(guint8) ((most_high_addr >> 8) & 0xff);
+				fw_index += 4;
+				break;
+			case 4:
+				addr_mode = 4;
+				most_high_addr = (guint16) hex_str_to_dec ((const gchar*) &hex_hdr[fw_index], 4);
+				checksum = line_len + rec_type +
+					(guint8) (start_addr & 0xff) +
+					(guint8) ((start_addr >> 8)&0xff) +
+					(guint8) (most_high_addr & 0xff) +
+					(guint8) ((most_high_addr >> 8) & 0xff);
+				fw_index += 4;
+				break;
+			default:
+				return FALSE;
+			}
+			if (end_of_file)
+				break;
+			checksum = -checksum;
+			if (checksum != (guint8) hex_str_to_dec ((const gchar*) &hex_hdr[fw_index], 2))
+				fw_index += 2;
+			return FALSE;
+		}
+		else {
+			fw_index += 2;
+			if (hex_hdr[fw_index] == 0x0D || hex_hdr[fw_index] == 0x0A)
+				fw_index ++;
+			else
+				new_line = TRUE;
+		}
+	}
+	/* only OCM */
+	if (img_header->fw_payload_len == 0 && img_header->fw_start_addr != 0)
+		img_header->fw_payload_len = fw_max_addr-fw_start_addr + last_len;
+	if (img_header->secure_tx_start_addr != 0 && img_header->secure_tx_payload_len == 0)
+		img_header->secure_tx_payload_len = fw_max_addr-fw_start_addr + last_len;
+	if (img_header->secure_rx_start_addr != 0 && img_header->secure_rx_payload_len == 0)
+		img_header->secure_rx_payload_len = fw_max_addr-fw_start_addr + last_len;
+	if (img_header->custom_start_addr != 0 && img_header->custom_payload_len == 0)
+		img_header->custom_payload_len = fw_max_addr-fw_start_addr + last_len;
+	img_header->total_len = img_header->fw_payload_len +
+		img_header->secure_tx_payload_len +
+		img_header->secure_rx_payload_len +
+		img_header->custom_payload_len;
+	g_debug ("total len:0x%x", img_header->total_len);
+	g_debug ("OCM start: 0x%x, len:0x%x", img_header->fw_start_addr,
+		 img_header->fw_payload_len);
+	g_debug ("Secure OCM TX start: 0x%x, len:0x%x",
+		 img_header->secure_tx_start_addr,
+		 img_header->secure_tx_payload_len);
+	g_debug ("Secure OCM RX start: 0x%x, len:0x%x",
+		 img_header->secure_rx_start_addr,
+		 img_header->secure_rx_payload_len);
+	g_debug ("Custom start: 0x%x, len:0x%x",
+		 img_header->custom_start_addr,
+		 img_header->custom_payload_len);

```

Comment:
If this ends up moving in to `fu-analogix-firmware.c` then I would think all this information should get ended up added into the `to_string` output.